### PR TITLE
style: remove lint-amnesty for dict literals

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** [ A short summary of what this PR does and why ]

JIRA: [AU-1064](https://2u-internal.atlassian.net/browse/AU-1064)

**What changed?**

- Remove old `lint-amnesty` for `dict-literal` updating to `{}` style.

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/openedx/edx-ora2/blob/master/.github/release_process.md)
- [x] Bumped version number in [setup.py](https://github.com/openedx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/setup.py#L39) and [package.json](https://github.com/openedx/edx-ora2/blob/a62e81a9b0d89223476967ec3c27f3557a850735/package.json#L3)

**Testing Instructions**

- Unit tests only, no behavior modified.

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

**Review by schenedx — APPROVED** [review]
👍 LGTM
